File: ecommerce-ts-express/packages/contracts/src/products/product.dto.ts
import { Currency } from '../common/currency.enum.js'

export type ProductDto = {
  id: string
  name: string
  price: string
  currency: Currency
  createdAt: string
  updatedAt: string
}

export type CreateProductDto = Omit<
  ProductDto,
  'id' | 'createdAt' | 'updatedAt'
>

File: ecommerce-go-fiber/shared/contracts/products/product_dto.go
package products

import "github.com/ivanmonichev/ecommerce-go-fiber/shared/contracts/common"

type ProductDTO struct {
	ID        string          `json:"id"`
	Name      string          `json:"name"`
	Price     string          `json:"price"`
	Currency  common.Currency `json:"currency"`
	CreatedAt string          `json:"createdAt"`
	UpdatedAt string          `json:"updatedAt"`
}

type CreateProductDTO struct {
	Name     string          `json:"name"`
	Price    string          `json:"price"`
	Currency common.Currency `json:"currency"`
}
